pkg/api/pod/util.go

```diff
@@ -832,6 +832,17 @@ func dropDisabledPodStatusFields(podStatus, oldPodStatus *api.PodStatus, podSpec
 		}
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ResourceHealthStatus) && !allocatedResourcesStatusInUse(podStatus) {
```

Comment:
I wonder if it isn't simpler AND faster to unconditionally clear the fields if the feature gate is off. We need to traverse the same data structures anyway, and the only price to pay would be to make `clear` safer, which seems a good thing anyway.

Reply:
done, makes sense

Reply:
thanks @SergeyKanzhelev . I only now notice that other feature gate checks seem do traverse the data structure anyway multiple times like you were doing. Not sure anymore which is the best approach, I'd be OK with whatever the API review folks suggest.